### PR TITLE
search jobs: add more tests for query validation

### PR DESCRIPTION
**Pull request by stefanhengl**
This improves our test coverage for `NewExhaustive`. The goal is to make sure we won't process unsupported or extremely expensive queries.

## Test plan
CI

### CODE REVIEW

File: internal/search/job/jobutil/exhaustive_job_test.go [hunk 1]
```diff
@@ -49,15 +51,64 @@ func TestNewExhaustive(t *testing.T) {
...
   (useFullDeadline . true)
   (patternInfo . TextPatternInfo{"content",re,nopath,filematchlimit:1000000})
   (numRepos . 1)
   (pathRegexps . [])
   (indexed . false))
 `),
-	}}
+		},
+		{
+			Name:  "",
```
Comment (keegancsmith) [inline internal/search/job/jobutil/exhaustive_job_test.go:64]: names?

File: internal/search/job/jobutil/exhaustive_job_test.go [hunk 2]
```diff
@@ -86,3 +138,52 @@ func TestNewExhaustive(t *testing.T) {
...
+		{query: `type:file index:no r:.* .*`, isPatterntypeRegex: true},
+		{query: `type:file index:no r:repo .*`, isPatterntypeRegex: true},
+		// predicates
+		{query: `type:file index:no repohasfile:foo.bar content`},
+		{query: `type:file index:no file:has.content("content")`},
+		{query: `type:file index:no repo:has.path("src") content`},
+	}
+
+	for _, c := range tc {
+		t.Run(c.query, func(t *testing.T) {
```
Comment (keegancsmith) [inline internal/search/job/jobutil/exhaustive_job_test.go:167]: minor: these produce very ugly test names that aren't that useful TBH. Maybe better to just be a bit more old school and produce an error message if there is no error from new exhaustive which includes the query string?

File: internal/search/job/jobutil/exhaustive_job_test.go [hunk 3]
```diff
@@ -86,3 +138,52 @@ func TestNewExhaustive(t *testing.T) {
...
+			patternType := query.SearchTypeLiteral
+			if c.isPatterntypeRegex {
+				patternType = query.SearchTypeRegex
+			}
```
Comment (keegancsmith) [inline internal/search/job/jobutil/exhaustive_job_test.go:171]: from reading the code, for exhaustive unless the string contains a patterntype in it, we will use patterntype standard.

**Review by keegancsmith — APPROVED** [review]
btw do the error messages we create here get shown to the user at all?

**Comment by stefanhengl** [comment]
> btw do the error messages we create here get shown to the user at all?

No. They are written to `failure_message` in the `exhaustive_search_jobs table`. We don't surface them anywhere. We could write them to the log as a stop-gap.

**Pull request merged**